runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java

```diff
@@ -1731,21 +1731,21 @@ private void doMergedInitialWindow(
                 initialMaxRW.value = Integer.MAX_VALUE;
                 produceStreams.forEach(p -> initialNoAckRW.value = Math.max(p.initialNoAck(), initialNoAckRW.value));
                 produceStreams.forEach(p -> initialPadRW.value = Math.max(p.initialPad, initialPadRW.value));
-                produceStreams.forEach(p -> initialMaxRW.value = Math.min(p.initialMax, initialMaxRW.value));
+                produceStreams.forEach(p -> initialMaxRW.value = Math.max(p.initialMax, initialMaxRW.value));
                 maxInitialNoAck = initialNoAckRW.value;
                 maxInitialPad = initialPadRW.value;
-                minInitialMax = initialMaxRW.value;
+                maxInitialMax = initialMaxRW.value;
```

Comment:
Not sure about this change.

The intention is to make sure we cannot overflow during produce even if we spend the flow control budget all on one partition.

If we go with this maxInitialMax change, then if partition 0 has 0 budget, but partition 1 has 64K budget, then we'd be able to send 64K to partition 0 on the merged stream which is an overflow.